src/jsrm/systems/planar_pcs_num.py

```diff
@@ -0,0 +1,1280 @@
+        l = params["l"] # length of each segment [m]
+        
+        # Classify the point along the robot to the corresponding segment
+        segment_idx, s_local, l_cum = classify_segment(params, s)
+            
+        chi_O = jnp.array([0.0, 0.0, th0])  # Initial pose of the robot
+
+        # Iteration function
+        def chi_i(
+            i: int, 
+            chi_prev: Array
+        ) -> Array:
+            th_prev = chi_prev[2]  # Extract previous orientation angle from val
+            p_prev = chi_prev[:2]  # Extract previous position from val
+            
+            # Extract strains for the current segment
+            kappa = xi[3 * i + 0]       # Bending strain
+            sigma_x = xi[3 * i + 1]     # Shear strain
+            sigma_y = xi[3 * i + 2]     # Axial strain
+            
+            # Compute the length of the current segment to integrate
+            l_i = jnp.where(i == segment_idx, s_local, l[i])
+            
+            # Compute the orientation angle for the current segment
+            dth = kappa * l_i  # Angle increment for the current segment
+            th = th_prev + dth
+
+            # Compute the integrals for the transformation matrix       
+            int_cos_th = ((jnp.sin(th) - jnp.sin(th_prev)) / kappa).reshape(())
+            int_sin_th = ((jnp.cos(th_prev) - jnp.cos(th)) / kappa).reshape(())
+            
+            # Transformation matrix
+            R = jnp.array([[int_cos_th, -int_sin_th],
+                   [int_sin_th, int_cos_th]])
+
+            # Compute the position
+            p = p_prev + (R @ jnp.array([sigma_x, sigma_y]).T).T
+            
+            return jnp.concatenate([p, jnp.array([th])])
+
+        chi, chi_list = lax.scan(
+            f = lambda carry, i: (chi_i(i, carry), chi_i(i, carry)),
+            init = chi_O, 
+            xs = jnp.arange(num_segments + 1))
+
+        return chi_list[segment_idx]
+
+    @jit
+    def forward_kinematics_fn(
+        params: Dict[str, Array], 
+        q: Array, 
+        s: Array, 
+        eps: float = global_eps
+        ) -> Array:
+        """
+        Compute the forward kinematics of the robot.
+
+        Args:
+            params (Dict[str, Array]): dictionary of robot parameters.
+            q (Array): configuration vector of the robot.
+            s (Array): point coordinate along the robot in the interval [0, L].
+            eps (float, optional): small number to avoid singularities. Defaults to global_eps.
+
+        Returns:
+            Array: pose of the robot at the point s in the interval [0, L].
+                The pose is represented as a 3D vector [x, y, theta] where (x, y) is the position
+                and theta is the orientation angle.
+        """        
+        # Map the configuration to the strains
+        xi = xi_eq + B_xi @ q
+                
+        # Add a small number to the bending strain to avoid singularities
+        xi = apply_eps_to_bend_strains(xi, eps)
```

Comment:
Please remove this as I think the new explicit implementation doesn't exhibit the singularities for `\kappa_be = 0` anymore, right?